sdk/keyvault/keyvault-common/src/challengeBasedAuthenticationPolicy.ts

```diff
@@ -156,25 +171,103 @@ export function createKeyVaultChallengeCallbacks(
 
     const accessToken = await getAccessToken([scope], {
       ...getTokenOptions,
+      enableCae: true,
       tenantId: parsedChallenge.tenantId,
     });
 
     if (!accessToken) {
-      return false;
+      // No access token provided, treat as no-op
+      return response;
     }
 
     request.headers.set("Authorization", `Bearer ${accessToken.token}`);
 
     challengeState = {
       status: "complete",
       scopes: [scope],
+      tenantId: parsedChallenge.tenantId,
+    };
+
+    // We have a token now, so try send the request again
+    return next(request);
+  }
+
+  async function handleCaeChallenge(
+    request: PipelineRequest,
+    response: PipelineResponse,
+    next: SendRequest,
+  ): Promise<PipelineResponse> {
+    // Cannot handle CAE challenge if a regular challenge has not been completed first
+    if (challengeState.status !== "complete") {
+      return response;
+    }
+
+    // If status is not 401, this is a no-op
+    if (response.status !== 401) {
+      return response;
+    }
+
+    const getTokenOptions = requestToOptions(request);
+
+    const challenge = response.headers.get("WWW-Authenticate");
+    if (!challenge) {
+      throw new Error("Missing challenge.");
```

Comment:
Can we improve this error? As a user of the SDK what information would I need to resolve any issues with 401 + missing challenge?